circuit/types/integers/src/mul_checked.rs

```diff
@@ -211,42 +211,52 @@ impl<E: Environment, I: IntegerType> Integer<E, I> {
 }
 
 impl<E: Environment, I: IntegerType> Metrics<dyn MulChecked<Integer<E, I>, Output = Integer<E, I>>> for Integer<E, I> {
-    type Case = (Mode, Mode);
+    type Case = (Mode, Mode, bool, bool);
```

Comment:
Is the `bool, bool` necessary?

Reply:
Removed this one ce85fac9c
the others don't seem as trivially removable.

Reply:
FYI the test has `bool, bool)` again, I must have reintroduced that because now the test case counts áre different depending on whether the values are zero or not.